.github/workflows/workflow.yml

```diff
@@ -0,0 +1,108 @@
+name: CI/CD Workflow
+
+on:
+  push:
+    branches: [master]
+
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "macos-latest"]
+    name: ${{ matrix.os }} Build
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash -l {0}
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Install `zstash` Package
+        run: |
+          python -m pip install --upgrade pip
+          pip install .
+
+      - name: Run Tests
+        run: |
+          cd tests
+          python -m unittest
+
+  publish-docs:
+    if: github.event_name == 'push'
+    needs: [build]
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Cache pip
+        uses: actions/cache@v2
+        with:
+          # This path is specific to Ubuntu
+          path: ~/.cache/pip
+          # Look to see if there is a cache hit for the corresponding requirements file
+          key: ${{ runner.os }}-pip-publish-docs
+          restore-keys: |
+            ${{ runner.os }}-pip-
+            ${{ runner.os }}-
+
+      - name: Install Dependencies
+        run: |
+          pip install sphinx==3.5.1 sphinx_rtd_theme==0.5.1 sphinx-multiversion==0.2.4
+
+      - name: Build Sphinx Docs
+        run: |
+          cd docs
+          sphinx-multiversion source _build/html
+
+      - name: Copy Docs and Commit
+        run: |
+          # gh-pages branch must already exist
+          git clone https://github.com/E3SM-Project/zstash.git --branch gh-pages --single-branch gh-pages
+
+          cd gh-pages
+          rm -r *
+
+          touch .nojekyll
+          # Add index.html to point to `master` branch automatically
+          printf '<meta http-equiv="refresh" content="0; url=./_build/html/master/index.html" />' >> index.html
+
+          cp -r ../docs/_build .
+          # Old builds must be siblings to current build to correctly load static content such as images and custom fonts
+          cp -r ../docs/_build_old .
+
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+
+          # The below command will fail if no changes were present, so we ignore it
+          git add .
+          git commit -m "Update documentation" -a || true
```

Comment:
Could it fail for another reason? We wouldn't want to ignore it then.

Reply:
I ported part of the code and that comment from a reference point.

It should be "ignore" rather than "fail" (e.g. when you do `git add .`) but no changes to the docs are detected so nothing is actually added.

Will update.

Reply:
It doesn't actually fail, so I'll remove that part of the comment (it was ported from a reference point).
For example, when you run `git add .` but no changes to the docs are detected so nothing is actually added.